src/ui.cpp

```diff
@@ -6,437 +6,577 @@
+				vout << YELLOW_(entry.first) << ", ";
+		}
+
+		vout << ")";
+		vout << std::endl;
 	}
-	vout << ")";
-	vout << std::endl;
-}
 
-void printErrorDetails(uint32_t error, const std::map<std::string, uint8_t>& errorMap)
-{
-	vout << "	(";
-	if (error == 0)
+	void printParameterOutOfBounds(std::string category, std::string field)
 	{
-		vout << GREEN("ALL OK") << ")" << std::endl;
-		return;
+		vout << "Motor config parameter in category [" << category << "] named [" << field
+			 << "] is out of bounds!" << std::endl;
 	}
 
-	for (auto& entry : errorMap)
+	void printFailedToSetupMotor(mab::Md80Reg_E regId)
 	{
-		if (error & (1 << entry.second) && entry.first[0] == 'E')
-			vout << RED_(entry.first) << ", ";
-		else if (error & (1 << entry.second) && entry.first[0] == 'W')
-			vout << YELLOW_(entry.first) << ", ";
+		vout << "Failed to setup motor! Error while writing: 0x" << std::hex
+			 << static_cast<uint16_t>(regId) << " register" << std::endl;
 	}
 
-	vout << ")";
-	vout << std::endl;
-}
-
-void printParameterOutOfBounds(std::string category, std::string field)
-{
-	vout << "Motor config parameter in category [" << category << "] named [" << field << "] is out of bounds!" << std::endl;
-}
-
-void printFailedToSetupMotor(mab::Md80Reg_E regId)
-{
-	vout << "Failed to setup motor! Error while writing: 0x" << std::hex << static_cast<uint16_t>(regId) << " register" << std::endl;
-}
-
-bool getDiffrentConfigsConfirmation(std::string configName)
-{
-	vout << "[MDTOOL] The default " << configName << " config was modified." << std::endl;
-	vout << "[MDTOOL] Would you like to revert it to default before downloading? [Y/n]" << std::endl;
-	char x;
-	std::cin >> x;
-	if (x != 'Y')
+	bool getDifferentConfigsConfirmation(std::string configName)
 	{
-		vout << "[MDTOOL] Using modified config." << std::endl;
-		return false;
+		vout << "[MDTOOL] The default " << configName << " config was modified." << std::endl;
+		vout << "[MDTOOL] Would you like to revert it to default before downloading? [Y/n]"
+			 << std::endl;
+		char x;
+		std::cin >> x;
+		if (x != 'Y' && x != 'y')
+		{
+			vout << "[MDTOOL] Using modified config." << std::endl;
+			return false;
+		}
+		vout << "[MDTOOL] Using default config." << std::endl;
+		return true;
 	}
-	vout << "[MDTOOL] Using default config." << std::endl;
-	return true;
-}
 
+	bool getUpdateConfigConfirmation(std::string configName)
+	{
+		vout << "[MDTOOL] The " << configName << " config is not complete." << std::endl;
+		vout << "[MDTOOL] Would you like to update lacking fields with default values before "
+				"downloading? [Y/n]"
+			 << std::endl;
+		char x;
+		std::cin >> x;
+		if (x != 'Y' && x != 'y')
+		{
+			vout << "[MDTOOL] Using unchanged user's config." << std::endl;
+			return false;
+		}
+		vout << "[MDTOOL] Updating config with default values." << std::endl;
+		vout << "[MDTOOL] New confing saved under name: [config_original_name]_updated.cfg."
```

Comment:
we know `[config_original_name]` here, it should be printed instead of placeholder